sshd/zenith/sshd/tunnel.py

```diff
@@ -373,36 +380,33 @@ def consul_heartbeat(
         elif liveness_failures > 0:
             # We want services to stay in the critical state until they succeed at least once
             status = "warning" if liveness_succeeded_once else "critical"
-    print(f"[SERVER] [INFO] Updating service health status to '{status}' in Consul")
+    logger.debug(f"Updating Consul service health to '{status}'")
     # Post the service information to consul
     response = requests.put(
         f"{server_config.consul_url}/v1/agent/check/update/{tunnel.id}",
         json = { "Status": status }
     )
     if 200 <= response.status_code < 300:
-        print("[SERVER] [INFO] Service health updated successfully")
+        logger.info(f"Updated Consul service health to '{status}'")
         # Reset the Consul failures on success
         return 0, liveness_failures, liveness_succeeded_once
     else:
         # If we failed to update the health status, emit a warning
-        print("[SERVER] [WARNING] Failed to update service health", file = sys.stderr)
+        logger.warning("Failed to update Consul service health")
     # Increment the failures if less than the limit
     if consul_failures < server_config.consul_heartbeat_failures:
         return consul_failures + 1, liveness_failures, liveness_succeeded_once
     # Otherwise, exit the process with an error status
-    print(
-        f"[SERVER] [ERROR] Failed to post status to Consul after {consul_failures} attempts",
-        file = sys.stderr
-    )
-    sys.exit(1)
+    logger.critical(f"Failed to update Consul service health after {consul_failures} attempts")
 
 
-def register_signal_handlers(server_config, tunnel):
+def register_signal_handlers(server_config, logger, tunnel):
     """
     Registers signal handlers for each of the exit signals we care about.
     """
     def signal_handler(signum, frame):
-        consul_deregister_service(server_config, tunnel)
+        consul_deregister_service(server_config, logger, tunnel)
+        logger.info("Tunnel disconnected by client")
```

Comment:
Will this log with enough context to know what happened?

Reply:
We can’t really know what happened, other than it was the client that hung up not us.